include/sundials/impl/sundials_errors_impl.h

```diff
@@ -0,0 +1,183 @@
+/* -----------------------------------------------------------------
+ * SUNDIALS Copyright Start
+ * Copyright (c) 2002-2022, Lawrence Livermore National Security
+ * and Southern Methodist University.
+ * All rights reserved.
+ *
+ * See the top-level LICENSE and NOTICE files for details.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ * SUNDIALS Copyright End
+ * -----------------------------------------------------------------
+ * !!!!!!!!!!!!!!!!!!!!!!!!! WARNING !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+ * This is a 'private' header file and should not be used in user
+ * code. It is subject to change without warning.
+ * !!!!!!!!!!!!!!!!!!!!!!!!! WARNING !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+ * -----------------------------------------------------------------
+ * Contains all error checking macros and private error handling API.
+ * -----------------------------------------------------------------*/
+
+#ifndef _SUNDIALS_ERRORS_IMPL_H
+#define _SUNDIALS_ERRORS_IMPL_H
+
+#include <sundials/sundials_errors.h>
+
+/* Shortcut macro to access the SUNContext declared with SUNAssignSUNCTX. */
+#define SUNCTX sunctx_
+
+/* The SUNAssignSUNCTX macro is used to declare the SUNContext
+   object to be used a function. */
+#if !defined(SUNDIALS_DISABLE_ERROR_CHECKS)
+#define SUNAssignSUNCTX(sunctx) \
+  SUNContext SUNCTX = sunctx; \
+  (void)SUNCTX
+#else
+#define SUNAssignSUNCTX(sunctx) \
+  SUNContext SUNCTX = sunctx;   \
+  (void)SUNCTX
+#endif
+
+/* ----------------------------------------------------------------------------
+ * Error checking macros
+ * ---------------------------------------------------------------------------*/
+
+/* SUNCheck is like SUNAssert except it is compiled even in release mode
+   unless SUNDIALS_DISABLE_ERROR_CHECKS is defined  */
+#if !defined(SUNDIALS_DISABLE_ERROR_CHECKS)
+#define SUNCheck(expr, code)                                                  \
```

Comment:
I might be misremembering, but I thought `SUNCheck` would return an error. I'm thinking of the example of checking a `malloc` call with `SUNCheck` rather than `SUNAssert` or was it that we would add a `SUNCheckNull` that returns?